### PR TITLE
[ADD] member_card: add member card report

### PR DESCRIPTION
**Pull request by tfrancoi**
This module add the ability to print the member card
from the contact to be use with beesdoo_base

### CODE REVIEW

**Comment by tfrancoi** [comment]
TODO: Remove polln specific logo
Add a field on the res.company to store the logo on the card

**Review by robinkeunen — APPROVED** [review]
added readme and todo

**Comment by carmenbianca** [comment]
I'm a little confused. This PR adds `polln_base`, not `member_card`.

**Comment by robinkeunen** [comment]
Indeed, that's why it is labelled with "TODO" : thibault still needs to extract the member card functionnalities to a new module out of polln_base.

**Comment by polchampion** [comment]
@remytms Est-ce que tu pourras estimer ceci (pas urgent) ? (avec le todo logo)

Virg et Pol : option pour l'intercoop, besoin d'une estimation pour avancer

**Comment by victor-champonnois** [comment]
Note : I've extracted member_card from beesdoo_base in this PR https://github.com/beescoop/Obeesdoo/pull/384 
@polchampion I can work on adding this member_card report on top of the beesdoo_base split if that's OK

**Comment by victor-champonnois** [comment]
~~@polchampion is this tested and OK to merge ?~~

**Comment by victor-champonnois** [comment]
branch rebased

**Comment by victor-champonnois** [comment]
This conflicts with https://github.com/beescoop/Obeesdoo/pull/384

**Comment by robinkeunen** [comment]
work was started here : https://github.com/beescoop/Obeesdoo/pull/432

**Comment by victor-champonnois** [comment]
- integrated the functionality in `member_card` and `eater_member_card` module. 
- The report in `eater_member_card` replaces the one in `member_card`, because I did not find a way to inherit it and add the `partner.parent_eater_id.name`, because of the foreach loop. Cf this [discussion](https://framateam.org/coopiteasy/pl/4gqyq6jswtrg3czwq5g3477pjh) and this [comment](https://github.com/beescoop/Obeesdoo/blob/d107373cf25ded5868688d0c5684acf778fd7f4c/eater_member_card/report/member_card_template.xml#L12)
- the commit [dc2bc07](https://github.com/beescoop/Obeesdoo/pull/289/commits/dc2bc07c93e91fbc95eb63ab77e11dfcf7507f1e) is not related to the task, it is a quick fix en passant.

The [Odoo task ](https://gestion.coopiteasy.be/web#id=9089&model=project.task&view_type=form&menu_id=)

File: eater_member_card/report/member_card_template.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+
+    <record id="member_card.report_member_card" model="ir.actions.report">
+        <field name="report_name">eater_member_card.member_card_template</field>
+    </record>
+
+
+
+
```
Comment (robinkeunen) [inline eater_member_card/report/member_card_template.xml]: too many blank lines

File: member_card/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -6,9 +6,8 @@ Create a member card and link it to a partner.
 - the partner's card and barcode history is visible in the member card tab
 - the barcode of a partner is computed from the last active member card
 - a boolean "Print Member card?" allows to filter on partners for which you need to print new cards.
+- contain a template for the member card
```
Comment (robinkeunen) [inline member_card/readme/DESCRIPTION.rst]:
```suggestion
- contains a generic template for the member card
```

File: eater_member_card/report/member_card_template.xml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    <template id="member_card_template">
+        <!-- replace the member_card template to add worker information.
+            Cannot use template inheritance because the worker information is in partner variable,
+        which is set in the foreach loop -->
```
Comment (robinkeunen) [inline eater_member_card/report/member_card_template.xml]: I'm surprised by this assertion, it works in other report inheritance 🤔

**Review by robinkeunen — CHANGES_REQUESTED** [review]

File: eater_member_card/report/member_card_template.xml (hunk 3 above)
Comment (victor-champonnois) [reply eater_member_card/report/member_card_template.xml]: I could not make it work, do you have an example ?

File: eater_member_card/report/member_card_template.xml (hunk 1 above)
Comment (victor-champonnois) [reply eater_member_card/report/member_card_template.xml]: fixed

File: eater_member_card/report/member_card_template.xml (hunk 3 above)
Comment (victor-champonnois) [reply eater_member_card/report/member_card_template.xml]: @robinkeunen up :)

**Comment by robinkeunen** [comment]
@victor-champonnois can you have a look at my changes ? J'ai un peu craqué mon slip :o

**Comment by robinkeunen** [comment]
@polchampion this was deployed on test server. Can you test it functionally  ?

`member_card`
```
Create a member card and link it to a partner.

- each member card has a barcode
- create new cards in the "member card" tab of the partner form
- use "Force Barcode" field to set a user defined barcode
- the partner's card and barcode history is visible in the member card tab
- the barcode of a partner is computed from the last active member card
- a boolean "Print Member card?" allows to filter on partners for which you need to print new cards.
- contains a generic template for the member card
- the card template uses field `member_card_logo` rather than the company logo

The wizards "Request member card printing" and "Set member card as printed" allow to

Careful : this module overrides the barcodes already defined on the partners.

If point of sale is installed, the generated barcode matches customer pattern rule.
```

`eater_member_card`
```
- eater barcode is computed from parent's barcode.
- Print Member card? is set to True when a new eater is created.
```

`eater_parent_code`
```
Adds field parent_barcode. This legacy field is necessary to generate
member card report with Jaspersoft.

This module depends on eater_member_card to generate the partner barcodes.

Do not migrate to 16.0
```

File: eater_member_card/models/partner.py [hunk 4]
```diff
@@ -1,31 +1,28 @@
...
-                    if c.valid:
-                        partner.barcode = c.barcode
+        parents = self.filtered(lambda p: not p.parent_eater_id)
+        eaters = self.filtered(lambda p: p.parent_eater_id)
+        super(Partner, parents)._compute_bar_code()
+
+        # warning : there is not constraint checking that parent_eater_id
+        # is set if eater field is set to eater
+        for partner in eaters:
+            partner.barcode = partner.parent_eater_id.barcode
```
Comment (victor-champonnois) [inline eater_member_card/models/partner.py:25]:
```suggestion
        parents = self.filtered(lambda p: not p.parent_eater_id)
        super(Partner, parents)._compute_bar_code()
        
        eaters = self.filtered(lambda p: p.parent_eater_id)
        # warning : there is not constraint checking that parent_eater_id
        # is set if eater field is set to eater
        for partner in eaters:
            partner.barcode = partner.parent_eater_id.barcode
```
Just reordering lines to make the operations clearer

**Review by victor-champonnois — APPROVED** [review]
Thanks for the changes :) 
@polchampion Virginie is the reference analyst on this task (cf [task](https://gestion.coopiteasy.be/web#id=9089&action=479&model=project.task&view_type=form&menu_id=536)), I ask her to test.

File: eater_member_card/models/partner.py (hunk 4 above)
Comment (robinkeunen) [reply eater_member_card/models/partner.py:25]: 👍

**Review by robinkeunen — APPROVED** [review]

**Pull request merged**